atd-etl/app/process/helpers_import.py

```diff
@@ -122,10 +145,22 @@ def generate_gql(line, fieldnames, file_type):
         fields = generate_fields_with_filters(line=line,
                                               fieldnames=fieldnames,
                                               filters=filters)
-
-        template = generate_template(name=query_name,
-                                     function=function_name,
-                                     fields=fields)
+        # Clean field list
+        template_fields = [field.lower() for field in fieldnames if field.lower() not in filters[0][1]]
```

Comment:
I'm out of the loop so I'm not sure why the fields need to be lowered. Could you help me understand more? Are the headers in the .csv uppercase?

Reply:
Yes, there are a couple reasons for this.

1. For some reason Hasura does not like camel-case for my fields 🤷‍♂it throws an error.
2. There are configurations settings containing column names in lower case in this script, some comparisons are made against those column names, and python is case sensitive when comparing strings. For example:

`filters[0][1]` contains all the needed to be removed.
`[field.lower() for field in fieldnames if field.lower() not in filters[0][1]]` in plain english: give me a lower-case array of all strings in fieldnames as long as they are not in the removed list.

Reply:
Thank you for your questions, I should make the code easier to read  or at least add some commentary

Reply:
Thanks for answering them - still familiarizing myself with the whole picture so this was very helpful!